scripts/BBoxManage.sh

```diff
@@ -0,0 +1,474 @@
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+  fi
+}
+
+function restore_hostapd_config () {
+  # Restore the original configuration file
+  if [ -e "$HOSTAPD_CONFIG.original" ]; then
+    if [ $DEBUG == 1 ]; then
+      echo 'Restoring $HOSTAPD_CONFIG from $HOSTAPD_CONFIG.original'
+    fi
+
+    cp $HOSTAPD_CONFIG.original $HOSTAPD_CONFIG 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    else
+      restart_hostapd
+    fi
+  fi
+}
+
+function backup_password_config () {
+  # Backup the original configuration file
+  if [ ! -e "$PASSWORD_CONFIG.original" ]; then
+    if [ $DEBUG == 1 ]; then
+      echo 'Backing up $PASSWORD_CONFIG to $PASSWORD_CONFIG.original'
+    fi
+
+    cp $PASSWORD_CONFIG $PASSWORD_CONFIG.original 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+
+    chmod 660 $PASSWORD_CONFIG.original 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+
+    chown biblebox:biblebox $PASSWORD_CONFIG.original 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+  fi
+}
+
+function restore_password_config () {
+  # Backup the original configuration file
+  if [ -e "$PASSWORD_CONFIG.original" ]; then
+    if [ $DEBUG == 1 ]; then
+      echo 'Restoring $PASSWORD_CONFIG from $PASSWORD_CONFIG.original'
+    fi
+
+    cp $PASSWORD_CONFIG.original $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    else
+      reload_nginx
+    fi
+  fi
+}
+
+function reset () {
+  restore_password_config
+  restore_hostapd_config
+  restore_nginx_config
+}
+
+function success () {
+  echo $SUCCESS
+  exit 0;
+}
+
+function failure () {
+  echo $FAILURE
+  exit 1;
+}
+
+function reload_nginx () {
+  # Restart the nginx service
+  if [ $DEBUG == 1 ]; then
+    echo "Reloading nginx configuration"
+  fi
+
+  /etc/init.d/nginx reload
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function unmountusb () {
+  # Unmount usb
+  if [ $DEBUG == 1 ]; then
+    echo "Unmounting USB media"
+  fi
+
+  pumount /media/usb0 2>&1 | logger -t $(basename $0)
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doshutdown () {
+  # Shutdown
+  if [ $DEBUG == 1 ]; then
+    echo "Shutting down"
+  fi
+
+  /sbin/shutdown now 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doreboot () {
+  # Reboot
+  if [ $DEBUG == 1 ]; then
+    echo "Rebooting"
+  fi
+
+  /sbin/reboot 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function restart_hostapd () {
+  # Restart the hostapd service
+  if [ $DEBUG == 1 ]; then
+    echo "Restarting hostapd"
+  fi
+
+  /etc/init.d/hostapd restart
```

Comment:
I am getting `interface wlan0 not configured` when I run `ifdown wlan0`. Is that because I am running in virtualbox? I do see wlan0 in /etc/network/interfaces.